instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/CassandraTest.java

```diff
@@ -0,0 +1,113 @@
+import io.opentelemetry.javaagent.instrumentation.apachecamel.RetryOnAddressAlreadyInUse;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import org.apache.camel.CamelContext;
+import org.apache.camel.ProducerTemplate;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.boot.SpringApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.testcontainers.containers.CassandraContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+public class CassandraTest extends RetryOnAddressAlreadyInUse {
+
+  @RegisterExtension
+  public static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static ConfigurableApplicationContext server;
+
+  @Container
+  private static final CassandraContainer<?> cassandra =
+      new CassandraContainer<>("cassandra:3.11.2").withExposedPorts(9042);
+
+  private static String host;
+
+  private static Integer port;
+
+  private static CqlSession cqlSession;
+
+  @BeforeAll
+  public static void setUp() {
+    withRetryOnAddressAlreadyInUse(CassandraTest::setUpUnderRetry);
+  }
+
+  private static void cassandraSetup() {
+    cqlSession =
+        CqlSession.builder()
+            .addContactPoint(cassandra.getContactPoint())
+            .withLocalDatacenter(cassandra.getLocalDatacenter())
+            .build();
+
+    cqlSession.execute(
+        "CREATE KEYSPACE IF NOT EXISTS test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+    cqlSession.execute("CREATE TABLE IF NOT EXISTS test.users (id int PRIMARY KEY, name TEXT);");
+  }
+
+  public static void setUpUnderRetry() {
+    cassandra.start();
+    cassandraSetup();
+
+    port = cassandra.getFirstMappedPort();
+    host = cassandra.getHost();
+
+    SpringApplication app = new SpringApplication(CassandraConfig.class);
+    app.setDefaultProperties(ImmutableMap.of("cassandra.host", host, "cassandra.port", port));
+    server = app.run();
+  }
+
+  @AfterAll
+  public static void cleanUp() {
+    if (server != null) {
+      server.close();
+      server = null;
+    }
+    cqlSession.close();
+    cassandra.stop();
+  }
+
+  @Test
+  public void testCassandra() {
+    CamelContext camelContext = server.getBean(CamelContext.class);
+    ProducerTemplate template = camelContext.createProducerTemplate();
+
+    template.requestBody("direct:input", (Object) null);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasKind(SpanKind.INTERNAL)
+                        .hasNoParent()
+                        .hasAttribute(AttributeKey.stringKey("camel.uri"), "direct://input"),
```

Comment:
Nit: by convention, we usually static import the `AttributeKey` factory methods"
```suggestion
                        .hasAttribute(stringKey("camel.uri"), "direct://input"),
```